compiler/modules/bir/bir.bal

```diff
@@ -42,17 +42,30 @@ public type Label int;
 # A RegionIndex is an index of a region in the regions array.
 public type RegionIndex int;
 
+public type Function FunctionDefn | AnonFunction;
+
+public type FunctionBase record {|
+    FunctionDecl decl;
+    # Index of source part in which the definition occurs
+    int partIndex;
+    # The position of the definition
+    Position position;
+    readonly...;
+|};
+
 # The definition of a function.
 public type FunctionDefn readonly & record {|
     *ModuleDefn;
+    *FunctionBase;
     # Name within the module
     InternalSymbol symbol;
-    FunctionDecl decl;
+|};
 
-    # Index of source part in which the definition occurs
-    int partIndex;
-    # The position of the definition
-    Position position;
+public type AnonFunction readonly & record {|
+    *FunctionBase;
+    # The index of the function in the module
+    int index;
```

Comment:
Why do we need this in AnonFunction but not in FunctionDefn?

Reply:
For references I guess, but can't we just refer to it directly?

Reply:
Moved `index` into the `*FunctionBase`